### ISSUE TITLE
Incompatability with react-native-gesture-handler v2

### ISSUE THREAD

**Issue by jspizziri** [charpeni/react-native-url-polyfill#440]
## Description

If you attempt to install this package in a project using `react-native-gesture-handler@^2.0.0` it fails to work.

I created this issue based on my findings here: https://github.com/charpeni/react-native-url-polyfill/issues/369#issuecomment-1158897289

## Steps to Reproduce

As I said in the comment above I've started getting this after upgrading a package. However, when I attempted to reproduce it in a brand new RN install, it started showing up once I added this line to my `tsconfig.json`:

```
{
  "compilerOptions": {
    ...
    "typeRoots": [ "./types", " ./node_modules/types"]
  },
  ...
}
```

## Example Errors Message
```
error TS2554: Expected 0 arguments, but got 1.
       const url = new URL("https://github.com");
                                 ~~~
error TS2339: Property 'searchParams' does not exist on type 'URL'.
        url.searchParams.append('height', heightConstraint.toString());
                   ~~~~~~~~~~~~
```

etc.

**Comment by jspizziri** [comment]
@charpeni thanks for your feedback! really appreciate it.

So I might've been wrong. I'm able to reproduce this in my `lib` by simply adding a new version of `react-native-gesture-handler`.

However, I tried to create a reproducible example from a brand new RN project and I'm actually getting the issue by simply adding some configuration to the `tsconfig.json` file:

```js
{
  "compilerOptions": {
    ...
    "typeRoots": [ "./types", " ./node_modules/types"]
  },
  ...
}
```

This is super strange to me as again, my TS config file has had this since the first time I started using this package and I've never had an issue til I upgraded the package I mentioned. But now I'm getting it when I add something to a new project that I've been using forever. 🤷‍♂️
